### DISCUSSION TITLE
ProjectTo ConstructUsing

### DISCUSSION THREAD

**Discussion by r-ising** [LuckyPennySoftware/AutoMapper#4199] [Ideas]
Hello,

we create a wrong mapping like this:

```
c.CreateMap<DateTime?, DateOnly?>().ConstructUsing(x => x.HasValue ? DateOnly.FromDateTime(x.Value) : null);
```

`AssertConfigurationIsValid` will work without any problems, but if you use the mapping, you will get the following error:

```
Unable to cast object of type 'System.Linq.Expressions.FullConditionalExpression' to type 'System.Linq.Expressions.NewExpression'.
```

The reason is that the ProjectBuilder will cast it to "NewExpression" and its not a "NewExpression".

Normally its easy to check the lambda (like x.Body.NodeType need to be ExpressionType.New). Is it possible to check it on "AssertConfigurationIsValid"?

Here is the full example:
https://gist.github.com/r-ising/ef54da6ba4db8264550095a6be90c7e0

**Comment by lbargaoanu** [comment]
No, because you might use `Map` for the same types. It's possible to do smth like that for `CreateProjection`. It could throw when using smth other than a `new` expression. A PR is welcome.
